### ISSUE TITLE
Bug: Filter expression with external variables do not work anymore

### ISSUE THREAD

**Issue by thomas-neitmann** [pharmaverse/admiral#1734]
### What happened?

After changing `assert_filter_cond(enquo(filter))` to `assert_filter_cond(enexpr(filter))` in `restrict_derivation()` (and other places) some code in {admiralophtha} failed as reported by @manciniedoardo. The code in question referred to variables defined within the function. As `enexpr()` does not capture the environment of symbols these variables couldn't be resolved later on. I think it's a very valid use case to pass an expression like `PARAMCD %in% params` where `params` has be defined elsewhere to the `filter` argument of a function.

Long story short for `filter` we should revert to using `enquo()` instead of `enexpr()` for `filter`. Fyi @bundfussr.

**Comment by thomas-neitmann** [comment]
I thought this was a quick fix but after reverting `enexpr(filter)` to `enquo(filter)` a bunch of unit test fail. Please take a look at `1734_revert_to_enquo` and take over if you're intrested. Thanks!

**Comment by manciniedoardo** [comment]
Hi @bundfussr When I do that, I still get the following error:
```
Error in assert_filter_cond(enexpr(filter)) : 
  is_expression(arg) is not TRUE
```

**Comment by bundfussr** [comment]
Hi @manciniedoardo, could you provide the complete code? I updated several cases where it worked. I would like to find out what is different in your case.

**Comment by manciniedoardo** [comment]
@bundfussr The complete code is a little long to paste here. However you can reproduce the error in the branch 49_derive_var_bcvacritxfl of {admiralophtha}. The function `derive_var_bcvacritxfl` (see [here](https://github.com/pharmaverse/admiralophtha/blob/49_bcvacritxfl%40devel/R/derive_var_bcvacritxfl.R) has three instances where the code uses a filter condition of the structure  `PARAMCD %in% paramcds` , and at the moment if you run the example code under `@examples` an error is  thrown because the object  `paramcds` cannot be found. Even if you substitute those three instances with  `PARAMCD %in% !!paramcds%` then you get another error (which I mentioned in my previous comment). Thanks for your help 😄

**Comment by bundfussr** [comment]
@manciniedoardo , the error thrown by `assert_filter_cond()` is a bug. I have created https://github.com/pharmaverse/admiraldev/issues/221 to fix it. Once the corresponding pull request is merged, `filter = PARAMCD %in% !!paramcds & BASETYPE %in% !!basetype` should work.

**Comment by manciniedoardo** [comment]
Thanks @bundfussr , I will keep an eye on that issue/PR. Appreciate your support on this!

**Comment by bundfussr** [comment]
@manciniedoardo , we merged the PR, i.e., if you install the latest devel version of admiraldev, it should work now.

**Comment by bundfussr** [comment]
Fixed by https://github.com/pharmaverse/admiraldev/issues/221.
